openlibrary/plugins/upstream/mybooks.py

```diff
@@ -200,172 +357,67 @@ class MyBooksTemplate:
     }
 
     def __init__(self, username, key):
+        """The following is data required by every My Books sub-template (e.g. sidebar)"""
         self.username = username
         self.user = web.ctx.site.get('/people/%s' % self.username)
+
+        if not self.user:
+            raise render.notfound("User %s" % self.username, create=False)
+
+        self.is_public = self.user.preferences().get('public_readlog', 'no') == 'yes'
+        self.user_itemname = self.user.get_account().get('internetarchive_itemname')
+
+        self.me = accounts.get_current_user()
+        self.my_username = self.me and self.me.key.split('/')[-1]
+        self.is_my_page = self.me and self.me.key.split('/')[-1] == self.username
         self.key = key.lower()
+        self.sponsorships = []
+
         self.readlog = ReadingLog(user=self.user)
         self.lists = self.readlog.lists
-        self.counts = self.readlog.reading_log_counts
+        self.counts = (
+            self.readlog.reading_log_counts
+            if (self.is_my_page or self.is_public)
+            else []
+        )
 
-    def render(
-        self,
-        page=1,
-        sort='desc',
-        list=None,
-        q="",
-        doc_count: int = 0,
-        ratings=None,
-        year=None,
-    ):
+        if self.me and self.is_my_page and self.key in self.ALL_KEYS:
+            self.counts.update(PatronBooknotes.get_counts(self.username))
+            self.sponsorships = get_sponsored_editions(self.user)
+            self.counts['sponsorships'] = len(self.sponsorships)
+
+        self.component_times = {}
+
+    def render_sidebar(self):
+        return render['account/sidebar'](
+            self.username,
+            self.key,
+            self.is_my_page,
+            self.is_public,
+            self.counts,
+            self.lists,
+            self.component_times,
+        )
+
+    def render_list(self, lst):
+        """Used from type/lists/view.html to render my books treatment"""
+        header_title = lst.get('name', 'List')
+        template = render["type/list/view_body"](lst, is_owner=self.is_my_page)
+        return self.render(template=template, header_title=header_title, page=lst)
```

Comment:
This method is only used once ; and from this context it's a little unclear what it does. Dissolve.